chain/types/ethtypes/errors.go

```diff
@@ -0,0 +1,44 @@
+package ethtypes
+
+import (
+	"fmt"
+
+	"golang.org/x/xerrors"
+)
+
+var ErrExecutionReverted = xerrors.New("execution reverted")
+
+// ExecutionRevertedError is an error that occurs when a transaction reverts.
+type ExecutionRevertedError struct {
+	Message string
+	Code    int
+	Meta    []byte
```

Comment:
Don't see this `Meta` field being used anywhere. It's neither written to or read from. Should be accepting it in the constructor and setting it ?

Reply:
We can remove this as we have Data field now.